### ISSUE TITLE
Add plots based on seaborn?

### ISSUE THREAD

**Issue by MichaelClerx** [pints-team/pints#589]
Seaborn makes matplotlib plots look nicer, and describes itself as "a library for making statistical graphics_", which "_aims to make visualization a central part of exploring and understanding data_".

It can do nice swarm plots and violin plots and other things useful for visualising parameter distributions.

Its dependencies are:

```
INSTALL_REQUIRES = [
    'numpy>=1.9.3',
    'scipy>=0.14.0',
    'pandas>=0.15.2',
    'matplotlib>=1.4.3',
]
```

which ups the numpy version and adds pandas (which has a bunch of dependencies of its own).

The other main downside of seaborn - the fact that it changes default matplotlib output - has been removed in recent releases: https://stackoverflow.com/questions/33099348/seaborn-configuration-hides-default-matplotlib
This was the main reason we didn't use seaborn when we started on Pints.

So I still don't like the added dependencies*, but am now a bit more open to the idea :D

**Comment by MichaelClerx** [comment]
Off-topic:

*I've been thinking about my dislike for dependencies in libraries a bit more, and have (at least partially) rationalised it as follows: 

1. It makes it harder to use the library on older systems, which is important when you want to use e.g. university clusters or other compute resources academics may have available. 
2. Every new dependency reduces workload for the programmer (and test writer), but introduces new work for the maintainer. Looking at github issue pages in general, bugs seem relatively rare (perhaps they're fixed faster). Similarly with Myokit, and what I gather from Chaste, a lot more time is spent maintaining compatibility with different versions of the dependencies than on fixing bugs in the local code

**Comment by mirams** [comment]
I normally dislike extra dependencies too. But perhaps surprisingly I have not much problem with including `pandas` and `seaborn`. 

pandas is pretty widely used and useful for reading in data files from all sorts of places. Even I managed to figure out how to get it working fairly fast when we used it with https://github.com/mirams/PyHillFit and the same for `seaborn` even when it was messing up matplotlib settings...
